storage/gcs/src/test/java/io/aiven/kafka/tieredstorage/storage/gcs/GcsStorageConfigTest.java

```diff
@@ -99,44 +106,57 @@ void emptyPathCredentials() {
     }
 
     @Test
-    void jsonAndPathCredentialsSet() {
+    void allCredentialsNull() {
         final var props = Map.of(
-            "gcs.bucket.name", "bucket",
-            "gcs.credentials.default", "false",
-            "gcs.credentials.json", "json",
-            "gcs.credentials.path", "path");
+            "gcs.bucket.name", "bucket"
+        );
         assertThatThrownBy(() -> new GcsStorageConfig(props))
             .isInstanceOf(ConfigException.class)
-            .hasMessage("gcs.credentials.json and gcs.credentials.path cannot be set together");
+            .hasMessage("All gcs.credentials.default, gcs.credentials.json, and gcs.credentials.path "
+                + "cannot be null simultaneously.");
     }
 
-    @Test
-    void jsonAndDefaultCredentialsSet() {
-        final var props = Map.of(
-            "gcs.bucket.name", "bucket",
-            "gcs.credentials.json", "json",
-            "gcs.credentials.default", "true");
+    @ParameterizedTest
+    @MethodSource("provideMoreThanOneCredentialsNonNull")
+    void moreThanOneCredentialsNonNull(final Boolean defaultCredentials,
+                                       final String credentialsJson,
+                                       final String credentialsPath) {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("gcs.bucket.name", "bucket");
+
+        if (defaultCredentials != null) {
+            props.put("gcs.credentials.default", Boolean.toString(defaultCredentials));
+        }
+        if (credentialsJson != null) {
+            props.put("gcs.credentials.json", credentialsJson);
+        }
+        if (credentialsPath != null) {
+            props.put("gcs.credentials.path", credentialsPath);
+        }
```

Comment:
And I have a suspicion that we are actually testing exactly the same things twice, are not we?

Reply:
> I think putting key with null value into configs does not change anything so I guess if you remove the ifs and replace `Boolean.toString` with just value it will still work exactly the same way.

Yes, this works.

Reply:
> And I have a suspicion that we are actually testing exactly the same things twice, are not we?

Pretty much, with different exception messages.